surface/include/pcl/surface/impl/mls.hpp

```diff
@@ -180,140 +180,40 @@ pcl::MovingLeastSquares<PointInT, PointOutT>::computeMLSPointNormal (int index,
-  // Compute the curvature surface change
-  if (curvature != 0)
-    curvature = fabsf (float (eigen_value / double (curvature)));
-
-
-  // Get a copy of the plane normal easy access
-  Eigen::Vector3d plane_normal = model_coefficients.head<3> ();
-  // Vector in which the polynomial coefficients will be put
-  Eigen::VectorXd c_vec;
-  // Local coordinate system (Darboux frame)
-  Eigen::Vector3d v_axis (0.0f, 0.0f, 0.0f), u_axis (0.0f, 0.0f, 0.0f);
-
-
-
-  // Perform polynomial fit to update point and normal
-  ////////////////////////////////////////////////////
-  if (polynomial_fit_ && static_cast<int> (nn_indices.size ()) >= nr_coeff_)
-  {
-    // Update neighborhood, since point was projected, and computing relative
-    // positions. Note updating only distances for the weights for speed
-    std::vector<Eigen::Vector3d, Eigen::aligned_allocator<Eigen::Vector3d> > de_meaned (nn_indices.size ());
-    for (size_t ni = 0; ni < nn_indices.size (); ++ni)
-    {
-      de_meaned[ni][0] = input_->points[nn_indices[ni]].x - point[0];
-      de_meaned[ni][1] = input_->points[nn_indices[ni]].y - point[1];
-      de_meaned[ni][2] = input_->points[nn_indices[ni]].z - point[2];
-      nn_sqr_dists[ni] = static_cast<float> (de_meaned[ni].dot (de_meaned[ni]));
-    }
-
-    // Allocate matrices and vectors to hold the data used for the polynomial fit
-    Eigen::VectorXd weight_vec (nn_indices.size ());
-    Eigen::MatrixXd P (nr_coeff_, nn_indices.size ());
-    Eigen::VectorXd f_vec (nn_indices.size ());
-    Eigen::MatrixXd P_weight; // size will be (nr_coeff_, nn_indices.size ());
-    Eigen::MatrixXd P_weight_Pt (nr_coeff_, nr_coeff_);
-
-    // Get local coordinate system (Darboux frame)
-    v_axis = plane_normal.unitOrthogonal ();
-    u_axis = plane_normal.cross (v_axis);
-
-    // Go through neighbors, transform them in the local coordinate system,
-    // save height and the evaluation of the polynome's terms
-    double u_coord, v_coord, u_pow, v_pow;
-    for (size_t ni = 0; ni < nn_indices.size (); ++ni)
-    {
-      // (Re-)compute weights
-      weight_vec (ni) = exp (-nn_sqr_dists[ni] / sqr_gauss_param_);
-
-      // Transforming coordinates
-      u_coord = de_meaned[ni].dot (u_axis);
-      v_coord = de_meaned[ni].dot (v_axis);
-      f_vec (ni) = de_meaned[ni].dot (plane_normal);
-
-      // Compute the polynomial's terms at the current point
-      int j = 0;
-      u_pow = 1;
-      for (int ui = 0; ui <= order_; ++ui)
-      {
-        v_pow = 1;
-        for (int vi = 0; vi <= order_ - ui; ++vi)
-        {
-          P (j++, ni) = u_pow * v_pow;
-          v_pow *= v_coord;
-        }
-        u_pow *= u_coord;
-      }
-    }
-
-    // Computing coefficients
-    P_weight = P * weight_vec.asDiagonal ();
-    P_weight_Pt = P_weight * P.transpose ();
-    c_vec = P_weight * f_vec;
-    P_weight_Pt.llt ().solveInPlace (c_vec);
-  }
-
-  if (cache_mls_results_)
```

Comment:
Do I understand right that caching became unconditional?

Reply:
It is conditional. Since the the vector gets resized I just fill out the object when it is passed to the function instead of constructing a new object within the computeMLSPointNormal method.

[Location 1](https://github.com/Levi-Armstrong/pcl/blob/bb532955896c706c43d3ea4a18ef5e2dd560e823/surface/include/pcl/surface/impl/mls.hpp#L371-L374)

[Location 2](https://github.com/Levi-Armstrong/pcl/blob/bb532955896c706c43d3ea4a18ef5e2dd560e823/surface/include/pcl/surface/impl/mls.hpp#L136-L143)

Reply:
Ah, right. Thanks for clarification.